content/derisking/state-field-guide/2-basic-principles.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Basic principles of modern software design
+layout: layouts/page
+permalink: /derisking/state-field-guide/basic-principles/
+sidenav: true
+sticky_sidenav: true
+tags: derisking
+eleventyNavigation:
+  key: derisking-state-basics
+  parent: derisking-state
+  order: 2
+  title: Basic principles of modern software design
+subnav:
+- text: User-centered design
+  href: '#user-centered-design'
+- text: Agile software development
+  href: '#agile-software-development'
+- text: Product ownership
+  href: '#product-ownership'
+- text: DevOps
+  href: '#devops'
+- text: Building with loosely coupled parts
+  href: '#building-with-loosely-coupled-parts'
+- text: Modular contracting
+  href: '#modular-contracting'
+- text: Footnotes
+  href: '#footnotes'
+---
+# State Software Budgeting Handbook {.usa-sr-only}
```

Comment:
Because our page template adds an `h1` based on the title, there is going to be a duplicate `h1`. If we stick closest to the current guide (with the lift-and-shift mindset), the title should become an `h2`. One possible solution I have been thinking about is to add a front-matter option named something like `custom-title`. Then add an `unless` condition around the `h1` to the page template. So when `custom-title: true` the page won't add the title from the front-matter, and we could manually add the `h1`/`#` and `h2`/`##`in markdown files (pages 2-4). This will also allow us to add a style rule for the h2s to have the top rule as in the current guide, and may also help us deal with page title vs tab title.

Reply:
Before starting to implement, I want to document my understanding:

1. The default behavior is that the `title` attribute gets used as the page's H1, via layouts/page#L19
1. If `custom-title` is true, then use the `title` attribute as the tab title, but it won't render the title in the page. For pages where `custom-title: true`, there's an expectation to add the titles (h1/h2) to the page manually.
1. For the derisking guide, we'll set `custom-title: true` on all pages, and then add page titles manually as H2s, in addition to this sr-only H1? (I'm least clear on this point.)

Reply:
You got it! The only minor quibble is on 3, the first page of both guides and the QASP page won't need `custom-title: true`. Also realizing we use underscores to separate frontmatter words so I guess it ought to be `custom_title`

Reply:
Moving this to #49